src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,40 +1,91 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { useAppDispatch, useAppSelector } from '../../app/hooks';
+import { getUser } from '../../api';
+import { User } from '../../types';
+import { actions } from '../../features/currentTodo';
 
 export const TodoModal: React.FC = () => {
+  const [user, setUser] = useState<User | null>(null);
+  const [hasError, setHasError] = useState(false);
+  const [isLoading, setIsLoading] = useState(true);
+
+  const dispatch = useAppDispatch();
+  const removeCurrentTodo = () => dispatch(actions.removeTodo());
+
+  const currentTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    const fetchUser = async () => {
+      try {
+        if (currentTodo) {
+          const userFromServer = await getUser(currentTodo.userId);
+
+          setUser(userFromServer);
+        }
+      } catch {
+        setHasError(true);
+      } finally {
+        setIsLoading(false);
+      }
+    };
+
+    fetchUser();
+  }, []);
+
+  if (!currentTodo) {
+    return null;
+  }
```

Comment:
You shouldn't render this component at all if there's no `currentTodo`. In your case, you waste resources to run code in `useEffect` and to get state from Redux, to declare some local state with `useState`. Just get `currentTodo` in the `<App />` and pass it here as a prop. But also render this component like this: `{currentTodo && <TodoModal currentTodo={currentTodo} />}`.